### PR TITLE
Upgrading libraries, mainly the JSqlParser, to support more queries out of the box.

### PR DESCRIPTION
**Pull request by onukristo**
## Context

Upgrading libraries, mainly the JSqlParser, to support more queries out of the box.

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/SqlParser.java [hunk 1]
```diff
@@ -4,64 +4,24 @@
...
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 import net.sf.jsqlparser.JSQLParserException;
 import net.sf.jsqlparser.parser.CCJSqlParser;
 import net.sf.jsqlparser.parser.StringProvider;
 import net.sf.jsqlparser.statement.Statements;
 
 public class SqlParser {
 
-  /*
```
Comment (onukristo) [inline tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/SqlParser.java:16]: New version supports the queries we tried to modify.

File: tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/TableAccessStatisticsSpyqlListener.java [hunk 2]
```diff
@@ -120,18 +127,26 @@ protected ParsedQuery parseSql(String sql, TwContext context) {
       return new ParsedQuery();
     }
 
-    ParsedQuery result = new ParsedQuery();
-    long startTimeMs = System.currentTimeMillis();
+    var result = new ParsedQuery();
```
Comment (tw-peeterkarolin) [inline tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/TableAccessStatisticsSpyqlListener.java:130]: Personally I don't see how using `var` here makes it better. `long` vs `var`, `ParsedQuery` vs `var`. In this case using `var` hurts the readability more in my opinion.

File: tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/TableAccessStatisticsSpyqlListener.java [hunk 3]
```diff
@@ -120,18 +127,26 @@ protected ParsedQuery parseSql(String sql, TwContext context) {
...
       var stmts = sqlParser.parse(sql, entryPointsProperties.getTas().getSqlParser().getTimeout());
 
-      for (Statement stmt : stmts.getStatements()) {
+      for (var stmt : stmts) {
+        if (stmt instanceof UnsupportedStatement) {
+          throw new IllegalStateException("Unsupported statement.");
+        }
+      }
+
+      for (Statement stmt : stmts) {
```
Comment (tw-peeterkarolin) [inline tw-entrypoints/src/main/java/com/transferwise/common/entrypoints/tableaccessstatistics/TableAccessStatisticsSpyqlListener.java:141]: Line 135:
`for (var stmt : stmts) {`
Line 141:
`for (Statement stmt : stmts) {`

?

**Review by tw-peeterkarolin — APPROVED** [review]

**Pull request merged**